Neos.ContentRepository.Core/Classes/Feature/Common/NodeVariationInternals.php

```diff
@@ -291,33 +269,20 @@ protected function collectNodePeerVariantsThatWillHaveBeenCreated(
      * except this operates on the to-be-varied node itself instead of an explicitly set succeeding sibling
      */
     private function resolveInterdimensionalSiblings(
-        ContentRepository $contentRepository,
-        ContentStreamId $contentStreamId,
+        ContentGraphInterface $contentGraph,
         NodeAggregateId $varyingNodeAggregateId,
         OriginDimensionSpacePoint $sourceOrigin,
         DimensionSpacePointSet $variantCoverage,
     ): InterdimensionalSiblings {
-        $originSubgraph = $contentRepository->getContentGraph()->getSubgraph(
-            $contentStreamId,
-            $sourceOrigin->toDimensionSpacePoint(),
-            VisibilityConstraints::withoutRestrictions()
-        );
-        $originSiblings = $originSubgraph->findSucceedingSiblingNodes(
-            $varyingNodeAggregateId,
-            FindSucceedingSiblingNodesFilter::create()
-        );
+        $originSiblings = $contentGraph
+            ->getSubgraph($sourceOrigin->toDimensionSpacePoint(), VisibilityConstraints::withoutRestrictions())
+            ->findSucceedingSiblingNodes($varyingNodeAggregateId, FindSucceedingSiblingNodesFilter::create());
 
         $interdimensionalSiblings = [];
         foreach ($variantCoverage as $variantDimensionSpacePoint) {
-            $variantSubgraph = $contentRepository->getContentGraph()->getSubgraph(
-                $contentStreamId,
-                $variantDimensionSpacePoint,
-                VisibilityConstraints::withoutRestrictions()
-            );
-
             // check the siblings succeeding in the origin dimension space point
             foreach ($originSiblings as $originSibling) {
-                $variantSibling = $variantSubgraph->findNodeById($originSibling->nodeAggregateId);
+                $variantSibling = $contentGraph->getSubgraph($variantDimensionSpacePoint, VisibilityConstraints::withoutRestrictions())->findNodeById($originSibling->nodeAggregateId);
```

Comment:
Just wondering: Why did you inline this?
I found the previous version easier to read and it didn't refetch the same instance in every loop

Reply:
Yeah no good reason, this is a leftover from the original Adapter change I think, were inlining made more sense.